cmd/hive/clustersync_test.go

```diff
@@ -0,0 +1,270 @@
+	assert.Contains(t, cmd.Example, "$ osdctl hive csf", "Command example should be set")
+
+	cmd.Flags().Set("limited-support", "true")
+	includeLimitedSupport := cmd.Flags().Lookup("limited-support").Value.String()
+	assert.Equal(t, "true", includeLimitedSupport, "Flag 'limited-support' should be parsed correctly")
+
+	cmd.Flags().Set("hibernating", "true")
+	includeHibernating := cmd.Flags().Lookup("hibernating").Value.String()
+	assert.Equal(t, "true", includeHibernating, "Flag 'hibernating' should be parsed correctly")
+
+	cmd.Flags().Set("syncsets", "false")
+	includeFailingSyncSets := cmd.Flags().Lookup("syncsets").Value.String()
+	assert.Equal(t, "false", includeFailingSyncSets, "Flag 'syncsets' should be parsed correctly")
+
+	cmd.Flags().Set("no-headers", "true")
+	noHeaders := cmd.Flags().Lookup("no-headers").Value.String()
+	assert.Equal(t, "true", noHeaders, "Flag 'no-headers' should be parsed correctly")
+
+	cmd.Flags().Set("output", "yaml")
+	output := cmd.Flags().Lookup("output").Value.String()
+	assert.Equal(t, "yaml", output, "Flag 'output' should be parsed correctly")
+
+	cmd.Flags().Set("cluster-id", "1234")
+	cmd.Flags().Set("limited-support", "true")
+	cmd.Flags().Set("hibernating", "false")
+	cmd.Flags().Set("syncsets", "true")
+	cmd.Flags().Set("output", "json")
+
+	err := cmd.Execute()
+	assert.NoError(t, err, "Command should execute without error")
+
+	err = cmd.Flags().Set("invalid-flag", "true")
+	assert.Error(t, err, "Setting invalid flag should return an error")
+}
+
+func testListFailingClusterSyncs(t *testing.T, returnErr error, expectedResults []failingClusterSync) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	mockClient := mockk8s.NewMockClient(mockCtrl)
+
+	setupMockClient(mockClient, returnErr)
+
+	options := &clusterSyncFailuresOptions{
+		kubeCli: mockClient,
+	}
+
+	result, err := options.listFailingClusterSyncs()
+
+	if returnErr != nil {
+		assert.Error(t, err)
+		assert.Nil(t, result)
+	} else {
+		assert.NoError(t, err)
+		assert.NotNil(t, result)
+		assert.Len(t, result, len(expectedResults))
+		for i, expected := range expectedResults {
+			assert.Equal(t, expected.Name, result[i].Name)
+			assert.Equal(t, expected.Namespace, result[i].Namespace)
+			assert.Equal(t, expected.LimitedSupport, result[i].LimitedSupport)
+			assert.Equal(t, expected.Hibernating, result[i].Hibernating)
+			assert.Contains(t, result[i].FailingSyncSets, expected.FailingSyncSets)
+			assert.Contains(t, result[i].ErrorMessage, expected.ErrorMessage)
+		}
+	}
+}
+
+func TestListFailingClusterSyncs(t *testing.T) {
+
+	expectedResults := []failingClusterSync{
+		{
+			Name:            "example-clustersync",
+			Namespace:       "uhc-production-1234",
+			Timestamp:       time.Now().Format(time.RFC3339),
+			LimitedSupport:  false,
+			Hibernating:     true,
+			FailingSyncSets: "selectorsyncset1 syncset1 ",
+			ErrorMessage:    "Failed to sync selectorsyncset1\n\nFailed to sync syncset1\n\n",
+		},
+	}
```

Comment:
Can you either add more failing `ClusterSync`s to test here or remove the `for` loop?

Reply:
added the test case as requested.